### PR TITLE
docs(Embed): update component's description, add example

### LINKED ISSUE #1330: Wrong description for Embed component
### Steps
Read through the JSDoc description of `Embed` component.

### Expected Result
Straightforward description of the `Embed` purpose should be provided

### Actual Result
Description for `Embed` states (and starts with) `A GIF is a muted segment of a video`, which is misleading. 

If primary role of `Embed` component is to host GIF, this should be, at the very least, reflected in the first sentence. Otherwise it seem like description is provided for the wrong component.

### Version
0.30.0

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #1330.

This PR:
- updates component's description including accessibility section
- adds example with `iframe`, Yandex maps are used because they have static image preview without API keys 🆓

### CODE REVIEW

File: docs/src/examples/components/Embed/Types/EmbedExample.shorthand.tsx [hunk 1]
```diff
@@ -6,7 +6,6 @@ const EmbedExample = () => (
     <Embed
       placeholder="https://raw.githubusercontent.com/bower-media-samples/big-buck-bunny-480p-5s/master/poster.jpg"
       video="https://raw.githubusercontent.com/bower-media-samples/big-buck-bunny-1080p-5s/master/video.mp4"
-      defaultActive={true}
```
Comment (layershifter) [inline docs/src/examples/components/Embed/Types/EmbedExample.shorthand.tsx:9]: This prop makes example a little bit strange, I expect that will be hidden by default

Comment (mnajdova) [reply docs/src/examples/components/Embed/Types/EmbedExample.shorthand.tsx:9]: Should we add example for it if we remove it from here?

File: docs/src/examples/components/Embed/Slots/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react'
+
+import ComponentExample from 'docs/src/components/ComponentDoc/ComponentExample'
+import ExampleSection from 'docs/src/components/ComponentDoc/ExampleSection'
+
+const Types = () => (
+  <ExampleSection title="Types">
+    <ComponentExample
+      title="IFrame"
+      description="A component can embed any iframe."
```
Comment (mnajdova) [inline docs/src/examples/components/Embed/Slots/index.tsx]:
```suggestion
      description="A Embed component can embed any iframe."
```

File: packages/react/src/components/Embed/Embed.tsx [hunk 3]
```diff
@@ -177,11 +177,10 @@ class Embed extends AutoControlledComponent<WithAsProp<EmbedProps>, EmbedState>
...
 /**
- * A GIF is a muted segment of a video.
- * @accessibility
- * If GIF should be visible to screen readers, textual representation needs to be provided in 'alt' or 'title' property.
+ * An embed component displays a placeholder and matching content based on user's interaction.
  *
- * Other considerations:
- *  - when alt and title property are empty, then Narrator in scan mode navigates to the gif and narrates it as empty paragraph
+ * @accessibility
+ * A `placeholder` slot represents an [`Image`](/components/image) component, please follow recommendations from its
```
Comment (mnajdova) [inline packages/react/src/components/Embed/Embed.tsx:183]: 👍

File: docs/src/examples/components/Embed/Slots/index.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react'
+
+import ComponentExample from 'docs/src/components/ComponentDoc/ComponentExample'
+import ExampleSection from 'docs/src/components/ComponentDoc/ExampleSection'
+
+const Types = () => (
+  <ExampleSection title="Types">
+    <ComponentExample
+      title="IFrame"
+      description="A Embed component can embed any iframe."
```
Comment (mnajdova) [inline docs/src/examples/components/Embed/Slots/index.tsx]: Sorry typo - An embed component...

**Review by mnajdova — APPROVED** [review]

File: docs/src/examples/components/Embed/Types/EmbedExample.shorthand.tsx (hunk 1 above)
Comment (layershifter) [reply docs/src/examples/components/Embed/Types/EmbedExample.shorthand.tsx:9]: Will add 👍

**Pull request merged**